docs/components/deferred.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Deferred
+new: true
+---
+
+## `<Deferred />`
+
+<details>
+  <summary>Type declaration</summary>
+
+```tsx
+export declare function Deferred<Data = any>({
+  children,
+  value,
+  fallback,
+  errorElement,
+}: DeferredProps<Data>): JSX.Element;
```

Comment:
Do we usually include referenced types (`DeferredProps`) in here as well?  If so the full chain down to primitives would be the following - but that feels a bit verbose.  Maybe we just add `DeferredProps` and if folks need to dig further into underlying types they can find them in the code?

```suggestion
export type Deferrable<T> = never | T | Promise<T>;
export type ResolvedDeferrable<T> = T extends null | undefined
  ? T
  : T extends Deferrable<infer T2>
  ? T2 extends Promise<infer T3>
    ? T3
    : T2
  : T;

export interface DeferredResolveRenderFunction<Data> {
  (data: ResolvedDeferrable<Data>): JSX.Element;
}

export interface DeferredProps<Data> {
  children: React.ReactNode | DeferredResolveRenderFunction<Data>;
  value: Data;
  fallbackElement: React.SuspenseProps["fallback"];
  errorElement?: React.ReactNode;
}

export declare function Deferred<Data = any>({
  children,
  value,
  fallback,
  errorElement,
}: DeferredProps<Data>): JSX.Element;
```

Reply:
Maybe we should have a TypeScript page like we do in the Remix docs 🤔

Reply:
might not be a bad idea, but I'm not sure we have much to fill it at the moment since there's no compiler concept to elaborate on.  At the least I think we can start with the simple approach you've got here (+ `DeferredProps`) and then add more thorough docs if we find folks are struggling?